harvard-university-for-the-creative-arts.csl

```diff
@@ -167,20 +203,25 @@
           </group>
         </group>
       </else-if>
-      <else-if type="map" match="any">
-        <group delimiter=" ">
-          <text variable="genre" prefix="[" suffix="]"/>
-          <text variable="scale" suffix="."/>
-          <text variable="publisher-place" suffix=":"/>
-          <text variable="publisher" suffix="."/>
-          <text variable="collection-title" prefix="(" suffix=")"/>
-        </group>
-      </else-if>
       <else>
-        <group delimiter=": ">
-          <text variable="publisher-place"/>
-          <text variable="publisher" suffix="."/>
-        </group>
+        <choose>
+          <if match="any" variable="publisher" type="motion_picture broadcast report musical_score song entry-dictionary entry-encyclopedia">
+            <text variable="publisher" suffix="."/>
+          </if>
+          <else>
+            <text value="s.n." prefix="(" suffix=")."/>
+          </else>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-place">
+    <choose>
+      <if match="any" variable="publisher-place" type="broadcast motion_picture report musical_score song entry-dictionary entry-encyclopedia manuscript">
+        <text variable="publisher-place"/>
+      </if>
```

Comment:
Thanks for your patience, I realise that what I was doing looked a bit confusing and I wasn’t explaining myself very well.

I have been had a thorough look at this, looking at the CSL-Zotero Mapping and at our referencing guide and what I am aiming to achieve is:

For the following CSL types: Thesis, Manuscript, Broadcast, article-newspaper
•	If the Place field is empty to not add s.l. as we don’t require a place field

For the following CSL types: Book, Entry-Dictionary, Entry-Encyclopedia, Motion Picture, Chapter, Speech, Song, Bill, Paper Conference, Report
•	If there is no url and the place field is blank I would like to see s.l. displayed
•	If there is a url and the place field is blank I would like to see nothing displayed (I wouldn’t like to see s.l.)
•	If there is a url and the place field is filled in I would like to see the place field displayed

Is this possible? Do you have any tips?


I really appreciate your help